### DISCUSSION TITLE
Problemi installazione zigbee2mqtt docker

### DISCUSSION THREAD

**Discussion by lucaferrocino** [indomus/forum#2125] [ZigBee, Z-Wave, Thread, Matter]
Ciao a tutti.
Non sto riuscendo ad installare zigbee2mqtt docker sul mio raspberry pi 4 con la chiavetta SONOFF Zigbee USB dongle più 3.0.
Ho aggiornato la chiavetta, seguito la guida  Installare e configurare ZigBee2MQTT con Docker su Raspberry Pi OS; una volta installato il docker zigbee2mqtt nei log mi esce il segiemte errore:

> Using '/app/data' as data directory
Zigbee2MQTT:info  2022-11-29 13:49:18: Logging to console and directory: '/var/log' filename: log.txt
Zigbee2MQTT:info  2022-11-29 13:49:18: Starting Zigbee2MQTT version 1.28.2 (commit #360a777)
Zigbee2MQTT:info  2022-11-29 13:49:18: Starting zigbee-herdsman (0.14.68)
Zigbee2MQTT:error 2022-11-29 13:49:19: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-11-29 13:49:19: Failed to start zigbee
Zigbee2MQTT:error 2022-11-29 13:49:19: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-11-29 13:49:19: Exiting...
Zigbee2MQTT:error 2022-11-29 13:49:19: Error: Error while opening serialport 'Error: Error: No such file or directory, cannot open /dev/ttyUSB.CC2531-01'
    at SerialPort.<anonymous> (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/uart.ts:90:28)
    at SerialPort._error (/app/node_modules/@serialport/stream/lib/index.js:198:14)
    at /app/node_modules/@serialport/stream/lib/index.js:242:12

ho seguito anche la guida per cambiare i permessi al /dev/ttyUSB0 ma continuo ad avere lo stesso problema. Ho usato anche direttamente la seguente stringa: /dev/serial/by-id/usb-Silicon_Labs_Sonoff_Zigbee_3.0_USB_Dongle_Plus_0001-if00-port0
ma continuo a non andare.
Aggiungo il codice docker-compose usato e la configurazione.yaml

>  zigbee2mqtt:
      container_name: zigbee2mqtt
      image: koenkk/zigbee2mqtt
      volumes:
      - /home/pi/zigbee2mqtt:/app/data
      - /run/udev:/run/udev:ro
      devices:
      - /dev/ttyUSB0
      environment:
      - TZ=Europe/Rome
      network_mode: host
      restart: always

>  homeassistant: true
    # Join alla rete MQTT disabilitato di default
    permit_join: false
    # MQTT settings
    mqtt:
      # MQTT base topic per i messaggi di zigbee2mqtt
      base_topic: zigbee2mqtt
      # MQTT server - URL (indirizzo LAN)
      server: 'mqtt://192.168.240.2'
      # MQTT server - Parametri di autenticazione (decommentare e popolare correttamente se si utilizzano credenziali per la connessione al broker MQTT
      user: pi
      password: Frontalino.11
      version: 5
      include_device_information: true
    serial:
      # Porta su cui è mappata l'antennza zigbee
      port: /dev/ttyUSB.CC2531-01
      adapter: ezsp 
    advanced:
      # Directory in cui verrà salvato il file di log
      log_directory: /var/log
    experimental: {}
    frontend:
      # porta su cui verrà esposta la webGUI
      port: 8083
    ota:
      update_check_interval: 10
    disable_automatic_update_check: false

Sapete dirmi dove sto sbagliando?
Grazie in anticipo

**Comment by KC-inDomus** [comment]
che stai sbagliando porta, e te lo dice anche chiaramente: "Error while opening serialport 'Error: Error: No such file or directory, cannot open /dev/ttyUSB.CC2531-01'"

**Reply by lucaferrocino** [reply]
Ho provato anche a seguire la guida che riporta nel log di errore [https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html#verify-that-the-user-you-run-zigbee2mqtt-as-has-write-access-to-the-port](https://github.com/indomus/forum/discussions/url) Anche con quella mi da errore.

**Reply by snakuzzo** [reply]
ma tu sei sicuro che la porta sia `/dev/ttyUSB.CC2531-01` ?
Mai vista una roba simile, sinceramente.
Hai usato `udev` per caso ?

**Reply by lucaferrocino** [reply]
si

**Reply by snakuzzo** [reply]
scusa eh...ma hai mappato il device quando hai creato il container ?
Perché dall'errore credo proprio di no.
Se sull'host la porta corretta è `/dev/ttyUSB.CC2531-01`, quando crei il container devi mappare quel device...

es. 

`--device /dev/ttyUSB.CC2531-01`   (se usi `docker run`)

oppure

```yaml
devices:
  - /dev/ttyUSB.CC2531-01:/dev/ttyUSB.CC2531-01
```

(se usi `docker-compose` o `docker compose`)

e poi nel `configuration.yaml` di zigbee2mqtt

```yaml
serial:
  port: /dev/ttyUSB.CC2531-01
```

**Comment by lucaferrocino** [comment]
adesso ho cancellato tutto e iniziato da capo. Funziona tutto tranne quando vado ad aggiungere l'integrazione su home assistant. Riporto il log:

> Using '/app/data' as data directory
Zigbee2MQTT:info  2022-11-29 16:00:40: Logging to console and directory: '/app/data/log/2022-11-29.16-00-40' filename: log.txt
Zigbee2MQTT:info  2022-11-29 16:00:40: Starting Zigbee2MQTT version 1.28.2 (commit #360a777)
Zigbee2MQTT:info  2022-11-29 16:00:40: Starting zigbee-herdsman (0.14.68)
Zigbee2MQTT:error 2022-11-29 16:01:12: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-11-29 16:01:12: Failed to start zigbee
Zigbee2MQTT:error 2022-11-29 16:01:12: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-11-29 16:01:12: Exiting...
Zigbee2MQTT:error 2022-11-29 16:01:13: TypeError: Cannot read properties of null (reading 'length')
    at AdapterNvMemory.init (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/adapter-nv-memory.ts:28:42)
    at ZnpAdapterManager.start (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/manager.ts:60:9)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/app/lib/zigbee.ts:58:27)
    at Controller.start (/app/lib/controller.ts:101:27)
    at start (/app/index.js:109:5)

Devo riavviare il raspberry per farlo rifunzionare ma non riesco ad associarlo ad home assistant

Queste sono le configurazioni che ho usate
Docker-compose:
> zigbee2mqtt:
    container_name: zigbee2mqtt
    restart: unless-stopped
    image: koenkk/zigbee2mqtt
    volumes:
      - /home/pi/zigbee2mqtt:/app/data
      - /run/udev:/run/udev:ro
    ports:
      - 8083:8083
    environment:
      - TZ=Europe/Rome
    devices:
      - /dev/ttyUSB0:/dev/ttyUSB0

configurazione.yaml
permit_join: true
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://192.168.240.2
serial:
  port: /dev/ttyUSB0
frontend:
  port: 8083
advanced:
  network_key:
    - 157
    - 204
    - 140
    - 38
    - 178
    - 233
    - 39
    - 64
    - 107
    - 20
    - 234
    - 216
    - 28
    - 152
    - 130
    - 70
  homeassistant_legacy_entity_attributes: false
  legacy_api: false
  legacy_availability_payload: false
device_options:
  legacy: false

**Reply by snakuzzo** [reply]
Ho guardato la tua configurazione...ma hai letto cosa ti ho scritto prima su come mappare la chiavetta ?

Poi...perché dovresti riavviare il RPi?
Non ha alcun senso questa cosa.

**Reply by snakuzzo** [reply]
Sicuro poi di avere l'ultimo firmware sulla CC2531 ?
Ricorda che è un hw vecchissimo e sconsigliato.
Inoltre con i fw precedenti dava problemi ed ogni volta la CC2531 andava scollegata e poi ricollegata prima di restartare Z2M

**Reply by lucaferrocino** [reply]
Si l'avevo letto, solo che già stavo facendo la prova partendo da zero.
Adesso funziona tutto mancava solo `homeassistant: true`nel file di configurazione.yaml
Grazie per l'aiuto e la disponibilità
